packages/apps/api/src/handlers/user.ts

```diff
@@ -264,63 +214,47 @@ export async function updateUser(
       modifiedFields.includes('lastName')) &&
     !userIsAdmin(request)
   ) {
-    response
-      .status(403)
-      .json(
-        toMessage(
-          'Attempt to modify immutable fields [' +
-            modifiedFields.join(',') +
-            ']'
-        )
-      );
-    return;
+    throw new HttpError({
+      status: 403,
+      message: `Attempt to modify immutable fields [${modifiedFields.join(
+        ','
+      )}]`,
+    });
   }
 
   if (modifiedFields.includes('email')) {
     const auth0EmailGet: APIResponse<Auth0Profile> = await auth0Client.getUserByEmail(
       fields.email.value
     );
-    if (auth0EmailGet.status !== ResponseStatus.NotFound) {
-      if (
-        auth0EmailGet.status === ResponseStatus.Success &&
-        auth0EmailGet.result.userId !== userId.toString()
-      ) {
-        response
-          .status(409)
-          .json(
-            toMessage(
-              'Auth0 user with email [' +
-                fields.email.value +
-                '] already exists'
-            )
-          );
-      } else if (auth0EmailGet.status === ResponseStatus.UnknownError) {
-        response.status(500).json(toMessage(auth0EmailGet.message));
+    if (auth0EmailGet.status === ResponseStatus.Success) {
+      if (auth0EmailGet.result.userId !== userId.toString()) {
+        throw new HttpError({
+          status: 409,
+          message: `Auth0 user with email [${fields.email.value}] already exists`,
+        });
       }
-      return;
+    } else if (auth0EmailGet.status !== ResponseStatus.NotFound) {
```

Comment:
Where do we handle the case where the auth0EmailGet returns a 404?

Reply:
We want it to return a 404!